grails-app/services/com/unifina/service/StorageNodeService.groovy

```diff
@@ -30,13 +32,23 @@ class StorageNodeService {
 	StreamStorageNode addStorageNodeToStream(EthereumAddress storageNodeAddress, String streamId) {
 		boolean exists = (StreamStorageNode.findByStorageNodeAddressAndStreamId(storageNodeAddress.toString(), streamId) != null)
 		if (!exists) {
-			StreamStorageNode instance = new StreamStorageNode(
-				streamId: streamId,
-				storageNodeAddress: storageNodeAddress.toString()
-			)
-			StreamStorageNode saved = instance.save(validate: true)
-			new NotifyStorageNodeTask(storageNodeAddress, streamId, NotifyStorageNodeTask.AssigmentEvent.STREAM_ADDED, streamService, streamrClientService).start()
-			return saved;
+			StreamStorageNode instance = new StreamStorageNode(streamService.getStream(streamId), storageNodeAddress.toString())
+			try {
+				StreamStorageNode saved = instance.save(validate: true, failOnError: true)
+				NotifyStorageNodeTask task = new NotifyStorageNodeTask(
+					storageNodeAddress,
+					streamId,
+					NotifyStorageNodeTask.AssigmentEvent.STREAM_ADDED,
+					streamService,
+					streamrClientService)
+				Thread thread = new Thread(task)
```

Comment:
Could modify `NotifyStorageNodeTask` to be a`Thread` instead of `Runnable` and set the name and uncaughtExceptionHandler in `NotifyStorageNodeTask's` constructor?

Reply:
It's recommended to implement `Runnable` instead of extending `Thread`. I.e. Composition over inheritance.

Reply:
Just trying to avoid duplicating the logic in lines 36-51 and 60-74. Moving those lines to a helper method would also work ok. Something like 
```
instance.save(validate: true, failOnError: true)
notifyStorageNode(streamId, STREAM_ADDED)
```

Reply:
It's a valid point, but I'll just smirk at it and wait for the third instance:
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)